plugin-server/src/worker/ingestion/team-manager.ts

```diff
@@ -160,9 +233,17 @@ export class TeamManager {
     private async syncPropertyDefinitions(properties: Properties, team: Team) {
         for (const [key, value] of Object.entries(properties)) {
             if (!this.propertyDefinitionsCache.get(team.id)?.has(key)) {
+                const isNumerical = typeof value === 'number'
+                const { propertyType, propertyTypeFormat } = detectPropertyDefinitionTypes(value, key)
```

Comment:
What's going to happen here, in the sql query below, and in `cacheEventNamesAndProperties` if the returned `propertyType` from is `null`?

Reply:
We store the nulls... e.g. https://github.com/PostHog/posthog/blob/b1613f6fa066710f7db9bf5d3641890a715783df/plugin-server/tests/worker/ingestion/team-manager.test.ts#L354

Reply:
Yup, but this was a "there's a bug in hinting at in this question" question. Have a look if the coDE ends up doing a lot more queries than it should

Reply:
This? https://github.com/PostHog/posthog/blob/b1613f6fa066710f7db9bf5d3641890a715783df/plugin-server/src/worker/ingestion/team-manager.ts#L327

That we would keep rechecking the nulls (even if we were never going to set a value for them)

Reply:
Yeah, I was basically hinting that we could also cache the existing property type for each property, and then run the sql insert either when there is no property, or the old type is null and the new type is set.

This does increase memory usage on the server somewhat, and the current implementation will only make one extra query per server restart per thread per property that's null, so maybe that's not too much... but with scale come edge cases, so...  🤔 .